includes/User.php

```diff
@@ -4146,7 +4152,14 @@ protected function loadOptions() {
 	 * @see getGlobalPreference for documentation about preferences
 	 */
 	protected function savePreferences() {
-		$this->userPreferences()->save($this->getId());
+		$additionalGlobalOptions = [];
+		foreach( $this->mOptions as $key => $value ) {
+			if ( $this->shouldOptionBeStored( $key, $value ) ) {
+				$additionalGlobalOptions[$key] = $value;
+			}
+		}
+
+		$this->userPreferences()->save($this->getId(), $additionalGlobalOptions);
```

Comment:
Are `$additionalGlobalOptions` necessary if you've already added all "private" properties to `$wgGlobalUserPreferenceWhiteList`?

Reply:
It's used to pass all needed private properties to save, in other case we will save only changed preferences and all read from user-preferences.